libs/nx-terraform/src/generators/create-environment-sp/generator.ts

```diff
@@ -0,0 +1,97 @@
+import { Tree } from '@nrwl/devkit'
+import execa from 'execa'
+import { isDryRun } from '../../common/isDryRun'
+import { readRepoSettings } from '../../common/read-repo-settings'
+import { readConfigFromEnvFile } from '../../common/readConfigFromEnvFile'
+import { NxTerraformCreateEnvironmentServicePrincipalSchema } from './schema'
+
+export default async function (
+    tree: Tree,
+    options: NxTerraformCreateEnvironmentServicePrincipalSchema,
+) {
+    const repoSettings = readRepoSettings()
+    const environmentConfig = await readConfigFromEnvFile(
+        repoSettings.terraformStateType,
+        options.environmentName,
+    )
+
+    const servicePrincipalName = `gh-actions-${repoSettings.azureResourcePrefix}-${options.environmentName}-sp`
```

Comment:
Would be good if there was a way to completely specify the name to aid in aligning with customer naming conventions - I've found they have them even for SP names

Reply:
Fixed, this is now an optional arg